hugr-core/src/hugr/rewrite/replace.rs

```diff
@@ -643,38 +643,56 @@ mod test {
     }
 
     #[test]
-    fn test_invalid() -> Result<(), Box<dyn std::error::Error>> {
+    fn test_invalid() {
+        let unknown_ext: ExtensionId = "unknown_ext".try_into().unwrap();
         let utou = FunctionType::new_endo(vec![USIZE_T]);
         let mk_op = |s| {
             CustomOp::new_opaque(OpaqueOp::new(
-                ExtensionId::new("unknown_ext").unwrap(),
+                unknown_ext.clone(),
                 s,
                 String::new(),
                 vec![],
                 utou.clone(),
             ))
         };
-        let mut h = DFGBuilder::new(FunctionType::new(
-            type_row![USIZE_T, BOOL_T],
-            type_row![USIZE_T],
-        ))?;
+        let mut h = DFGBuilder::new(
+            FunctionType::new(type_row![USIZE_T, BOOL_T], type_row![USIZE_T])
+                .with_extension_delta(unknown_ext.clone()),
+        )
+        .unwrap();
         let [i, b] = h.input_wires_arr();
-        let mut cond = h.conditional_builder(
-            (vec![type_row![]; 2], b),
-            [(USIZE_T, i)],
-            type_row![USIZE_T],
-        )?;
-        let mut case1 = cond.case_builder(0)?;
-        let foo = case1.add_dataflow_op(mk_op("foo"), case1.input_wires())?;
-        let case1 = case1.finish_with_outputs(foo.outputs())?.node();
-        let mut case2 = cond.case_builder(1)?;
-        let bar = case2.add_dataflow_op(mk_op("bar"), case2.input_wires())?;
-        let mut baz_dfg = case2.dfg_builder(utou.clone(), bar.outputs())?;
-        let baz = baz_dfg.add_dataflow_op(mk_op("baz"), baz_dfg.input_wires())?;
-        let baz_dfg = baz_dfg.finish_with_outputs(baz.outputs())?;
-        let case2 = case2.finish_with_outputs(baz_dfg.outputs())?.node();
-        let cond = cond.finish_sub_container()?;
-        let h = h.finish_hugr_with_outputs(cond.outputs(), &PRELUDE_REGISTRY)?;
+        let mut cond = h
+            .conditional_builder_exts(
+                (vec![type_row![]; 2], b),
+                [(USIZE_T, i)],
+                type_row![USIZE_T],
+                unknown_ext.clone(),
+            )
+            .unwrap();
+        let mut case1 = cond.case_builder(0).unwrap();
```

Comment:
Why `?` -> `unwrap` everywhere? It does make things a lot longer and look like you are changing more than you are.